### DISCUSSION TITLE
3D Printed Panel / Parts

### DISCUSSION THREAD

**Discussion by nebhead** [nebhead/PiFire#24] [Ideas]
I'm starting a new discussion thread on 3D Printed Panels / Parts for PiFire.  

I'm personally, just learning to use OnShape to create my own models.  My first order of business will be to create an enclosure for the existing Traeger controller on my Texas Grill, so that I can move that down to the bottom side and then utilize the front opening for PiFire.

**Comment by faisalhusain007** [comment]
Cool addition I can upload the sketches for what I have done here.

**Reply by nebhead** [reply]
Yep!  Thought that it would be good to start a thread here to post our designs as we get them finalized.

**Reply by faisalhusain007** [reply]
I have moved our conversation here with the 3D design elements.

I have reduced the gaps between the screen and the case window and chamfered the display window border by 5mm.

I think we could use a modeling silicon to create a gasket around the the display window and edges of the case to make it more dust proof and water resistant. 

![image](https://user-images.githubusercontent.com/36872476/109160075-4e81a200-77b0-11eb-86ba-0e3ee2cf9822.png)

This is a revision of the previous design, started printing will see what it looks like when finished.

**Reply by nebhead** [reply]
Very nice!  Curious how you are fastening the screen to the back of this print.  Since the display isn't completely flat on the front, and the screw holes are so tiny, I was thinking it might require screws to mount through the face plate with nuts behind the PCB screen.

**Comment by faisalhusain007** [comment]
I have two solutions the back of printed panels have 4 posts, one version
uses 2.5mm threaded columns where a small screen can be used to secure the
screen. However repeated tightening and loosening of the screws wear out
the thread. The second version the posts are chamfered by 1mm, this allows
the screen to be held in place at the correct height, then I use a small 3d
printed bracket that goes across the back of the LCD pcb and fixes to the
printed frame. I use this for myself as I am constantly removing the screen.

![16142770045627837661093061100084](https://user-images.githubusercontent.com/36872476/109198148-d1b6ee00-77d8-11eb-883c-61fb4f4c2c83.jpg)

**Comment by nebhead** [comment]
That makes sense.  Is the bracket pretty secure?  Or would that be
something you'd want to glue down for the final solution?

**Reply by faisalhusain007** [reply]
You can glue down but I prefer 3m all weather double sided foam tape.

**Comment by nebhead** [comment]
I spent a bunch of hours this weekend learning OnShape and putting together my own design for the 3.2 inch screen (w/buttons).  It's big... it's not the prettiest, but I'm learning as I go.  :)   Here is a preview.

![PiFire-Model](https://user-images.githubusercontent.com/19876886/109553732-3b105880-7a88-11eb-9e05-2ddca0980cdc.png)

This doesn't include mounting any of the other hardware (i.e. the PiFire v2 board and the relay board).  But it's a start.  The way this is designed, it's made up of four pieces that individually can be printed without supports.  In addition, my idea is that the front panel can be customized to support different screen sizes, touch/no-touch, buttons / no-buttons, etc.  And that piece can be printable in only a couple hours vs. printing up the entire thing.  

Anyway, my thought is that we can have multiple options for panels & hardware mounting boxes!  Choice is good.  :)

**Comment by faisalhusain007** [comment]
I like it, looks good, I do like the idea of modular system. That's kind of
what I was thinking separate the screen from the pcb and relays. Also for
the different types of screens /buttons/non buttons we could have blanking
plates that could be fitted.

**Comment by nebhead** [comment]
I finally got around to publishing the 3D models for printing here:  https://www.thingiverse.com/thing:4872825  

I kind of hastily posted these, so I'm sure I'm missing information, but hopefully the drawings are somewhat self explanatory.  

![pifire-assembly](https://user-images.githubusercontent.com/19876886/120079059-1a47a700-c067-11eb-8f86-751873b21cfc.png)

![pifire-explode](https://user-images.githubusercontent.com/19876886/120078969-b9b86a00-c066-11eb-9a47-aec451fa3f92.png)

**Comment by faisalhusain007** [comment]
Looks awesome cant wait to get back home to test it out.

**Reply by nebhead** [reply]
Oh no, are you still travelling?

**Comment by faisalhusain007** [comment]
Unfortunately I am stuck in London. I have not been able to travel back
because of the current situation. I hope to be back sometime soon.
